akvo/rsr/spa/app/modules/editor/actions.js

```diff
@@ -24,38 +24,40 @@ export const setStatus = (publishingStatus) => (dispatch, getState) => {
     .catch((error) => { dispatch({ type: actionTypes.BACKEND_ERROR, error, response: error.response.data }) })
 }
 export const fetchFields = (sectionIndex, fields) => ({ type: actionTypes.FETCH_SECTION, sectionIndex, fields })
-export const fetchSetItems = (sectionIndex, setName, items) => ({ type: actionTypes.FETCH_SET_ITEMS, sectionIndex, setName, items})
+export const fetchSetItems = (sectionIndex, setName, items) => ({ type: actionTypes.FETCH_SET_ITEMS, sectionIndex, setName, items })
 export const addSetItem = (sectionIndex, setName, item) => (dispatch, getState) => {
-  dispatch({ type: actionTypes.ADD_SET_ITEM, sectionIndex, setName, item})
+  dispatch({ type: actionTypes.ADD_SET_ITEM, sectionIndex, setName, item })
   const setItems = get(getState().editorRdr[`section${sectionIndex}`].fields, setName)
   const itemIndex = setItems.length - 1
   api.post(getEndpoint(sectionIndex, setName), item, getTransform(sectionIndex, setName, 'request'), null, true)
-    .then(({ data: {id, periods}}) => { dispatch({ type: actionTypes.ADDED_SET_ITEM, sectionIndex, setName, id, itemIndex, periods }) })
+    .then(({ data: { id, periods } }) => { dispatch({ type: actionTypes.ADDED_SET_ITEM, sectionIndex, setName, id, itemIndex, periods }) })
     .catch((error) => {
       dispatch({ type: actionTypes.BACKEND_ERROR, error, sectionIndex, setName: `${setName}[${setItems.length - 1}]`, response: error.response ? error.response.data : error, statusCode: error.response.status })
       dispatch({ type: actionTypes.REMOVED_SET_ITEM, failedAdd: true, sectionIndex, setName, itemIndex: setItems.length - 1, skipValidation: true })
     })
 }
 export const editSetItem = (sectionIndex, setName, itemIndex, itemId, fields) => (dispatch) => {
+  const inputFields = fields?.disaggregationTargets?.length > 0
+    ? { ...fields, disaggregationTargets: fields.disaggregationTargets.filter(item => (item?.value)) }
```

Comment:
I think it would be enough to make sure all the entries in the `disaggregation_targets` array have a `value` and set them to `null` when the `value` field is missing. We of course filter out items which are null in the array.